lib/sacastats_web/controllers/character_controller.ex

```diff
@@ -0,0 +1,66 @@
+defmodule SacaStatsWeb.CharacterController do
+  use SacaStatsWeb, :controller
+  alias SacaStatsWeb.CharacterView
+
+  def character(conn, %{"character_name" => name, "stat_type" => stat_template_name}) do
+    # Fetch character info from API and DB,
+
+    if String.downcase(stat_template_name) == "lookup" do
+      redirect(conn, to: Routes.character_path(conn, :character, name, "general"))
+    end
+
+    q =
+      PS2.API.Query.new(collection: "character")
+      |> PS2.API.QueryBuilder.term("name.first_lower", String.downcase(name))
+
+    {:ok, %PS2.API.QueryResult{data: body}} = PS2.API.query_one(q)
+    {:ok, %PS2.API.QueryResult{returned: returned}} = PS2.API.query_one(q)
+
+    if returned == 0 do
+      conn
+      |> put_flash(:error, "The character '" <> name <> "' doesn't appear to exist.")
+      |> redirect(to: Routes.character_path(conn, :character_search))
+    end
+
+    character_stuff = %{
+      "name" => name,
+      "stat_page" => String.downcase(stat_template_name) <> ".html",
+      "response" => Map.get(body, "character_id")
+    }
+
+    render(conn, "template.html", character: character_stuff)
+  end
```

Comment:
`case` statements are also a more common way to make decisions/branches.
```suggestion
    case PS2.API.query_one(q) do
      {:ok, %PS2.API.QueryResult{returned: 0}} ->
        conn
        |> put_flash(:error, "The character '" <> name <> "' doesn't appear to exist.")
        |> redirect(to: Routes.character_path(conn, :character_search))
      {:ok, %PS2.API.QueryResult{data: body}} ->

        character_stuff = %{
          "name" => name,
          "stat_page" => String.downcase(stat_template_name) <> ".html",
          "response" => Map.get(body, "character_id")
        }

        render(conn, "template.html", character: character_stuff)
      {:error, e} ->
        Logger.error("Error fetching character: #{inspect e}")
    end
```